src/lib/optimizer/strategy/abstract_rule.cpp

```diff
@@ -9,9 +9,74 @@
 
 namespace opossum {
 
+/**
+ * IMPORTANT NOTES ON OPTIMIZING SUBQUERY LQPS
```

Comment:
I'd move this into the header, as it is an important interface information, not an implementation detail. Also could you please update the comment with information about the new interface?